controlpanel/api/models.py

```diff
@@ -50,27 +53,51 @@ def aws_delete_role(self):
         services.delete_role(self.iam_role_name)
 
     def helm_create(self):
-        helm.init_user(self.username, self.email, self.get_full_name())
-        helm.config_user(self.username)
+        user_slug = sanitize_dns_label(self.username)
+        values = ",".join(f"{key}={value}" for key, value in {
+            "NFSHostname": settings.NFS_HOSTNAME,
+            "Username": user_slug,
+            "Email": self.email,
+            "Fullname": self.get_full_name(),
+            "Env": settings.ENV,
+            "OidcDomain": settings.OIDC_DOMAIN,
+        })
+        helm.upgrade_release(
+            f"init-user-{user_slug}",
+            "mojanalytics/init-user",
+            f"--set={values}",
+        )
+        helm.upgrade_release(
+            f"config-user-{user_slug}",
+            "mojanalytics/config-user",
+            "--namespace=user-{user_slug}",
+            "--set=Username={user_slug}",
+        )
```

Comment:
#### 1b. Update models

> Lines 55 - 75

I refactored our Helm client module and this calling code to fix the levels of abstraction. The module now only provides an interface to Helm. The model knows how to invoke Helm when creating a User.

[Next →]()